.github/workflows/build.yml

```diff
@@ -65,14 +65,34 @@ jobs:
       - name: Show disk usage before maven
         run: df -kh
 
-      - name: Build with Maven
-        timeout-minutes: 120
+      - name: Build with Maven (Linux)
+        if: runner.os == 'Linux'
+        timeout-minutes: 90
         shell: bash
         run: |
           ./mvnw clean verify -DredirectTestOutput=true \
           --show-version --batch-mode --errors --no-transfer-progress \
           -DtrimStackTrace=false \
           -Dsurefire.rerunFailingTestsCount=2
+
+      - name: Deploy Docker (Linux)
+        if: runner.os == 'Linux' && github.ref == 'refs/heads/trunk'
+        timeout-minutes: 90
+        shell: bash
+        run: |
+          ./mvnw deploy -Ddockerfile.username=$${{ secrets.DOCKERHUB_USER }} \
+           -Ddockerfile.password=... $${{ secrets.DOCKERHUB_TOKEN }}
+
```

Comment:
We shouldn't be publishing snapshot builds of trunk to DockerHub. Can we make sure we're only publishing released artifacts?